mmv1/third_party/terraform/services/datastream/resource_datastream_connection_profile_test.go

```diff
@@ -247,3 +359,156 @@ resource "google_datastream_connection_profile" "mysql_con_profile" {
 }
 `, context)
 }
+
+func testAccDatastreamConnectionProfile_sshKey_update(context map[string]interface{}, preventDestroy bool, private_key string, public_key string) string {
+	context["lifecycle_block"] = ""
+	if preventDestroy {
+		context["lifecycle_block"] = `
+        lifecycle {
+            prevent_destroy = true
+        }`
+	}
+	context["private_key"] = private_key
+	context["public_key"] = public_key
+
+	return acctest.Nprintf(`
+resource "random_password" "pwd" {
+	length = 16
+	special = false
+}
+
+resource "google_sql_database_instance" "instance" {
+    depends_on         = [google_compute_instance.default]
+    name            	= "tf-test-my-database-instance%{random_suffix}"
+    database_version	= "POSTGRES_14"
+    region           	= "us-central1"
+    settings {
+        tier = "db-f1-micro"
+        ip_configuration {
+			ipv4_enabled = true
+
+			authorized_networks {
+				value = google_compute_instance.default.network_interface.0.access_config.0.nat_ip
+			}
+        }
+    }
+    
+    deletion_protection  = "false"
+}
+
+resource "google_sql_database" "db" {
+	depends_on = [google_sql_database_instance.instance]
+	instance = google_sql_database_instance.instance.name
+	name     = "db"
+}
+resource "google_sql_user" "user" {
+	depends_on	= [google_sql_database_instance.instance]
+	name		= "user"
+	instance	= google_sql_database_instance.instance.name
+	password	= random_password.pwd.result
+}
+
+resource "google_compute_instance" "default" {
+	name         = "tf-test-instance-%{random_suffix}"
+	machine_type = "e2-small"
+	zone         = "us-central1-a"
+	boot_disk {
+		initialize_params {
+			image = "debian-11-bullseye-v20241009"
+		}
+	}
+
+	network_interface {
+		network    = "default"
+		access_config {}
+		}
+
+	metadata = {
+		"ssh-keys" = "user:%{public_key}"
+	}
+
+	metadata_startup_script = <<-EOT
+	#!/bin/bash
+	echo "Updating SSHD config for SSH forwarding..."
+
+	# Backup sshd_config
+	echo "AllowTcpForwarding yes" >> /etc/ssh/sshd_config
+	echo "PasswordAuthentication no" >> /etc/ssh/sshd_config
+	echo "PubkeyAuthentication yes" >> /etc/ssh/sshd_config
+	echo "AuthorizedKeysFile .ssh/authorized_keys" >> /etc/ssh/sshd_config
+	
+	# Restart SSH service
+	systemctl restart sshd
+	EOT
+
+	tags = ["ssh-host"]
+	
+}
+
+resource "time_sleep" "ssh_host_wait" {
+	depends_on = [google_compute_instance.default]
+	create_duration = "15m"
```

Comment:
You should experiment with this timeout value - make it long enough that the test passes reliably but not so long that it's annoying to wait for :-)